### PR TITLE
Add com.valvesoftware.Steam.CompatibilityTool.Proton

### PR DESCRIPTION
**Pull request by gasinvein**
This is Proton build on top of freedesktop-sdk.
Components that are available from FD.o SDK are omitted (e.g. gstreamer), otherwise it shouldn't be much different from official builds.

Originally, this was started as an experiment of building Proton flatpak without Steam Runtime and related tooling. But since 5.13, official Proton [stopped working](https://github.com/flathub/com.valvesoftware.Steam/issues/642) in flatpak due to dependency on Valve's own "app container" tool called pressure-vessel. Once (or if) it is resolved, this build might become less useful, but for now it may provide an option for users who want to use latest Proton with Steam flatpak.

Currently missing features (I consider them non-essential, as most of the games don't need it):
- Bundled fonts (would require significant amount of work for rather small gain)
- Custom gstreamer plugin (requires flathub/org.freedesktop.Sdk.Extension.rust-stable#20)

### CODE REVIEW

**Comment by Saroufim** [comment]
This build can be a stepping stone towards creating packages for custom proton builds such as GE.

**Comment by gasinvein** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Proton

**Comment by gasinvein** [comment]
@Saroufim Well, I personally didn't plan to flatpak customized Protons forks, but anyone interested in doing so is welcome to reuse this one.

**Comment by gasinvein** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Proton

**Comment by nanonyme** [comment]
Looks good to me.

File: com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>com.valvesoftware.Steam.CompatibilityTool.Proton</id>
+  <extends>com.valvesoftware.Steam</extends>
+  <name>Proton</name>
+  <summary>Compatibility tool for Steam Play based on Wine and additional components</summary>
```
Comment (LeandroStanger) [inline com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:6]: Add metainfo translation to pt_BR and es for Proton
```suggestion
  <summary>Compatibility tool for Steam Play based on Wine and additional components</summary>
  <summary xml:lang="pt_BR">Ferramenta de compatibilidade para Steam Play baseada no Wine e componentes adicionais</summary>
  <summary xml:lang="es">Herramienta de compatibilidad para Steam Play basada en Wine y componentes adicionales</summary>
```

Comment (gasinvein) [reply com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:6]: Hm, I guess the summary should indicate that it isn't official. Something like
>Compatibility tool based on Wine (unofficial flatpak build)

Comment (nanonyme) [reply com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:6]: Alternatively community build. The important thing to stress ia this isn't backed by Valve. This should also be made obvious when selecting tool within Steam

Comment (nanonyme) [reply com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:6]: I might even go as far as name "Proton (community build)

Comment (gasinvein) [reply com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:6]: So what should `name` and `summary` be like, exactly? My English is too limited to word it nicely.

Comment (nanonyme) [reply com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:6]: Looks like for the app this stuff is in description https://github.com/flathub/com.valvesoftware.Steam/blob/beta/com.valvesoftware.Steam.metainfo.xml#L12

Comment (gasinvein) [reply com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:6]: I've set tool's name to `Proton (community build)` both in metainfo and compatibilitytool.vdf. Let's hope it won't be too long for UI parts.

File: com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>com.valvesoftware.Steam.CompatibilityTool.Proton</id>
+  <extends>com.valvesoftware.Steam</extends>
+  <name>Proton (community build)</name>
+  <summary>Compatibility tool for Steam Play based on Wine and additional components</summary>
+  <summary xml:lang="pt_BR">Ferramenta de compatibilidade para Steam Play baseada no Wine e componentes adicionais</summary>
+  <summary xml:lang="es">Herramienta de compatibilidad para Steam Play basada en Wine y componentes adicionales</summary>
```
Comment (LeandroStanger) [inline com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:8]: Update metainfo translation to pt_BR and es for Proton (Community build)
```suggestion
  <name>Proton (Community build)</name>
  <name xml:lang="pt_BR">Proton (Construção da comunidade)</name>
  <name xml:lang="es">Proton (Construcción comunitaria)</name>
  <summary>
  <p>Compatibility tool for Steam Play based on Wine and additional components</p>
  <p>This is a community package of Steam Play plaform not officially supported by Valve. Report bugs through linked issue tracker.</p> 
  </summary>
  <summary xml:lang="pt_BR">
  <p>Ferramenta de compatibilidade para Steam Play baseada no Wine e componentes adicionais</p>
  <p>Este é um pacote comunitário da plataforma Steam Play não oficialmente suportado pela Valve. Relatar bugs por meio do rastreador de problemas vinculado.</p>
  </summary>
  <summary xml:lang="es">
  <p>Herramienta de compatibilidad para Steam Play basada en Wine y componentes adicionales</p>
  <p>Este es un paquete comunitario de plataforma Steam Play que Valve no admite oficialmente. Informar errores a través del rastreador de problemas vinculado.</p>
  </summary>
```

Comment (gasinvein) [reply com.valvesoftware.Steam.CompatibilityTool.Proton.metainfo.xml:8]: `summary` tag can't include markup and it shouldn't be this long.

**Comment by gasinvein** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Proton

**Pull request closed without merging**